xarray/tests/test_backends.py

```diff
@@ -2213,6 +2212,77 @@ class PyNioTestAutocloseTrue(PyNioTest):
     autoclose = True
 
 
+@requires_netCDF4
+@requires_pnc
+class TestPseudoNetCDF(CFEncodedDataTest, NetCDF3Only, TestCase):
+    def test_write_store(self):
+        # pseudonetcdf is read-only for now
+        pass
+
+    @contextlib.contextmanager
+    def open(self, path, **kwargs):
+        with open_dataset(path, engine='pseudonetcdf',
+                          autoclose=self.autoclose,
+                          **kwargs) as ds:
+            yield ds
+
+    def save(self, dataset, path, **kwargs):
+        dataset.to_netcdf(path, engine='netcdf4', **kwargs)
+
+    def test_weakrefs(self):
+        example = Dataset({'foo': ('x', np.arange(5.0))})
+        expected = example.rename({'foo': 'bar', 'x': 'y'})
+
+        with create_tmp_file() as tmp_file:
+            example.to_netcdf(tmp_file, engine='netcdf4')
+            on_disk = open_dataset(tmp_file, engine='pseudonetcdf')
+            actual = on_disk.rename({'foo': 'bar', 'x': 'y'})
+            del on_disk  # trigger garbage collection
+            assert_identical(actual, expected)
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle(self):
+        pass
+
+    @pytest.mark.skip(reason='cannot pickle file objects')
+    def test_pickle_dataarray(self):
+        pass
+
+    @pytest.mark.skip(reason='does not support boolean dtype')
+    def test_roundtrip_boolean_dtype(self):
+        # pnc does not support boolean dtype
```

Comment:
This is done by xarray's conventions handling logic. The underlying data is stored as integers. Unless you explicitly strip out attributes I think this should still work?

Reply:
On the other hand, maybe all these conventions aren't relevant for pseudonetcdf, since it's not designed to open arbitrary netcdf files?

Reply:
Same as above.

I am not stripping out attributes, but...

```
    def test_roundtrip_boolean_dtype(self):
        original = create_boolean_data()
        self.assertEqual(original['x'].dtype, 'bool')
        with self.roundtrip(original) as actual:
>           assert_identical(original, actual)
E           AssertionError: <xarray.Dataset>
E           Dimensions:  (t: 4)
E           Dimensions without coordinates: t
E           Data variables:
E               x        (t) bool True False False True
E           <xarray.Dataset>
E           Dimensions:  (t: 4)
E           Dimensions without coordinates: t
E           Data variables:
E               x        (t) int8 ...
```

Reply:
I agree that it is not designed to open arbitrary netcdf files. I could upload another test file (e.g., uamiv formatted) and disable this test... or just disable this test.